ovirtlago/virt.py

```diff
@@ -156,12 +157,15 @@ def get():
         except AssertionError:
             raise RuntimeError('Failed to connect to the engine')
 
-        return api.pop()
+        return api_v3.pop()
 
     def get_api(self):
-        if self._api is None or not self._api.test():
-            self._api = self._get_api()
-        return self._api
+        return self.get_api_v3()
+
+    def get_api_v3(self):
+        if self._api_v3 is None or not self._api_v3.test():
+            self._api_v3 = self._get_api_v3()
+        return self._api_v3
```

Comment:
I sincerely hope we are not going to have more major versions any time soon, so I think this is premature. I will add the exception though - makes sense.
I'll also actually add v4 API and test to see this is all not breaking...